pkg/rke2/rke2.go

```diff
@@ -57,23 +54,22 @@ func Agent(clx *cli.Context, cfg Config) error {
 
 func setup(clx *cli.Context, cfg Config) error {
 	var dataDir string
+
 	for _, f := range clx.Command.Flags {
 		switch t := f.(type) {
 		case cli.StringFlag:
 			if strings.Contains(t.Name, "data-dir") {
-				dataDir = t.Value
+				dataDir = *t.Destination
 			}
 		}
 	}
 
 	for _, f := range clx.App.Flags {
 		switch t := f.(type) {
 		case cli.StringFlag:
-			if t.Name == "profile" && t.Destination != nil && *t.Destination != "" {
-				cisMode = true
+			if strings.Contains(t.Name, "profile") {
```

Comment:
Changed to be consistent with clx.Command.Flags processing above.

Reply:
Would it be better to make sure it is equal?

Reply:
The way @briandowns set it up, the value is validated over at https://github.com/rancher/rke2/blob/master/pkg/cli/cmds/root.go#L148 so by the point we get here, if it's not an empty string then we want to turn on cisMode.